### PR TITLE
HTML template: Facilitate Specification Extensions

### PR DESCRIPTION
**Pull request by scottpopplewell**
Fixes https://github.com/asyncapi/generator/issues/136

### CODE REVIEW

File: templates/html/.partials/schema-prop.html [hunk 1]
```diff
@@ -89,6 +89,9 @@
             {% endfor %}
           </div>
         {% endif %}
+        {% for extensionName, extensionValue in prop.extensions() %}
+            <div class="text-sm markdown">{{ extensionName }}: {{ extensionValue }}</div>
```
Comment (fmvilas) [inline templates/html/.partials/schema-prop.html]: `extensionValue` can be huge here depending on the extension. I wonder if we should think about another way of visualizing it. Also, if it's a JSON object, doesn't it render `[Object object]`?

**Comment by scottpopplewell** [comment]
@fmvilas It is finished. Ready to be merged.

File: templates/html/.partials/schema-prop.html [hunk 2]
```diff
@@ -108,6 +108,34 @@
...
+                    <span class="text-sm" style="word-break: break-word;">{{ extensionName }}</span>
+                    <svg class="expand" version="1.1" viewBox="0 0 24 24" x="0"
+                      xmlns="http://www.w3.org/2000/svg" y="0">
+                      <polygon points="17.3 8.3 12 13.6 6.7 8.3 5.3 9.7 12 16.4 18.7 9.7 "></polygon>
+                    </svg>
+                  </div>
+                </div>
+                <div class="children">
+                  <div class="{% if odd %}bg-grey-lighter{% else %}bg-grey-lightest{% endif %} {% if not root %}pl-8 pr-8{% endif %} rounded">
+                    <pre class="text-sm">{{ extensionValue | prettyPrint }}</pre>
```
Comment (fmvilas) [inline templates/html/.partials/schema-prop.html]: You don't need a custom filter for "pretty printing". Instead use `dump(2)`.

```suggestion
                    <pre class="text-sm">{{ extensionValue | dump(2) }}</pre>
```

File: templates/html/.filters/all.js [hunk 3]
```diff
@@ -30,6 +31,14 @@ module.exports = ({ Nunjucks, Markdown, OpenAPISampler }) => {
...
     return Array.isArray(arr);
   });
+  
+  Nunjucks.addFilter('isObject', (obj) => {
+    return typeof obj === 'object' && obj !== null;
+  });
+
+  Nunjucks.addFilter('prettyPrint', (obj) => {
+    return JSON.stringify(obj, null, 4);
+  });
```
Comment (fmvilas) [inline templates/html/.filters/all.js]: As pointed out below, you'll not need this.

Comment (scottpopplewell) [reply templates/html/.filters/all.js]: Beautiful. Thanks for that. It's a great change. I've updated all.js and schema-prop.html

**Pull request merged**